feature-libs/asm/components/asm-main-ui/asm-main-ui.component.ts

```diff
@@ -53,6 +57,8 @@ export class AsmMainUiComponent implements OnInit, OnDestroy {
   customer$: Observable<User | undefined>;
   isCollapsed$: Observable<boolean> | undefined;
   iconTypes = ICON_TYPE;
+  customerIdInURL: string;
```

Comment:
Do we need to be setting global variables here? If so, we prefer to use Observables for variables to detect changes during code flow. It would be great if we can move the logic that defines these into a service. eg. `this.service.getIsEmulated(): Observable`.

Reply:
As the isEmulated is only used in this component, so maybe we can keep it in here, Anyway, I moved to amsComponentService